File: SortingAndSearching/Tickets.cpp
#include <iostream>
#include <bits/stdc++.h>
#include <vector>
using namespace std;
int binSearch(vector<long long> a, long long key){
    int l = 0;
    int r = a.size()-1;
    int ans = l;
    while(l <= r){
        int mid = (l + r)/2;
        if(a[mid] == key){
            return mid;
        }
        else if(a[mid] < key){
            ans = mid;
            l = mid+1;
        }
        else{
            r = mid-1;
        }
    }
    if(a[ans] <= key) return ans;
    return -1;
}
int main(){
    int n, m;
    cin >> n >> m;
    vector<long long> a(n);
    long long b[m];
    for(int i = 0 ; i < n; i++){
        long long x;
        cin >> x;
        a.push_back(x);
    }
    for(int i = 0 ; i < m; i++){
        cin >> b[i];
    }
    sort(a.begin(), a.end());
    for(int i = 0 ; i < m; i++){
        if(a.size()==0){
            cout << "-1";
            continue;
        }
        int x = binSearch(a, b[i]);
        if(x == -1){
            cout << "-1";
        }
        else{
            if(a[x] == 0){ 
                cout << "-1 "; continue;
            }
            cout << a[x] << endl;
            a.erase(a.begin() +x);
        }
    }


}

File: SortingAndSearching/Tickets.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

/**
 * Tickets
 */
public class Tickets {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        ArrayList<Long> a = new ArrayList<>();
        for(int i = 0 ; i < n; i++){
            a.add(i, sc.nextLong());
        }
        long[] b = new long[m];
        for(int i = 0 ; i < m; i++){
            b[i] = sc.nextLong();
        }
        Collections.sort(a);
        for(int i = 0 ; i < m; i++){
            if(a.size()==0){
                System.out.print("-1 ");
                continue;
            }
            int x = binSearch(a, b[i]);
            if(x == -1) System.out.print("-1 ");
            else{
                
                System.out.print(a.get(x) + " ");
                a.remove(x);
            }
        }
        System.out.println();

    }
    public static int binSearch(ArrayList<Long> a, long key){
        int l = 0;
        int r = a.size()-1;
        int ans = l;
        while(l <= r){
            int mid = (l + r)/2;
            if(a.get(mid) == key){
                return mid;
            }
            else if(a.get(mid) < key){
                ans = mid;
                l = mid+1;
            }
            else{
                r = mid-1;
            }
        }
        if(a.get(ans) <= key) return ans;
        return -1;
    }
}